syntax = "proto3";

package api.v1;

// 定义认证服务 (Service)
// 职责：处理所有与“我是谁”有关的请求
service AuthService {
  // 接口 1: 登录
  // 输入账号密码，如果正确，返回一个 Token (通行证)
  rpc Login (LoginReq) returns (LoginRes);

  // 接口 2: 注册
  // 创建新账号
  rpc Register (RegisterReq) returns (RegisterRes);
  
  // 接口 3: 校验 Token (这是给网关用的内部接口)
  // 网关收到 WebSocket 连接请求时，调用此接口检查 Token 是否合法
  rpc VerifyToken (VerifyTokenReq) returns (VerifyTokenRes);
}

// --- 数据结构定义 (Message) ---

// 登录请求
message LoginReq {
  string username = 1; 
  string password = 2; // 注意：实际传输建议用 HTTPS 加密，或者传输 Hash 后的值
}

// 登录响应
message LoginRes {
  bool success = 1;
  int64 user_id = 2;   // 用户唯一 ID (后续聊天全靠这个 ID 认人)
  string token = 3;    // JWT 或 SessionID (核心通行证)
  string error_msg = 4;
}

// 注册请求 (保持最简，仅需用户名密码)
message RegisterReq {
  string username = 1;
  string password = 2;
  // string email = 3; // 可选扩展
}

message RegisterRes {
  bool success = 1;
  int64 user_id = 2;
  string error_msg = 3;
}

// Token 校验请求
message VerifyTokenReq {
  string token = 1;
}

// Token 校验响应
message VerifyTokenRes {
  bool valid = 1;      // Token 是否有效
  int64 user_id = 2;   // 如果有效，告诉网关这个 Token 属于哪个 UserID
}
